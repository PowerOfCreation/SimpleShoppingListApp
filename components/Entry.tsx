import {
  GestureResponderEvent,
  TouchableOpacity,
  StyleSheet,
} from "react-native"
import { ThemedText } from "./ThemedText"
import React from "react"
import Feather from "@expo/vector-icons/Feather"

type EntryProps = {
  ingredientName: string
  isCompleted: boolean
  onToggleComplete: (event: GestureResponderEvent) => void
  onLongPress: (event: GestureResponderEvent) => void
}

export function Entry(props: EntryProps) {
  const getBackgroundColor = () => {
    return props.isCompleted
      ? styles.completedBackground
      : styles.uncompletedBackground
  }

  const getTextStyles = () => {
    return props.isCompleted ? [styles.completedText] : [styles.defaultText]
  }

  return (
    <TouchableOpacity
      style={[getBackgroundColor(), styles.buttonStyle]}
      onPress={props.onToggleComplete}
      onLongPress={props.onLongPress}
    >
      <Feather
        name={props.isCompleted ? "check-circle" : "circle"}
        size={26}
        color="#1999b3"
      />

      <ThemedText
        style={[styles.baseText, getTextStyles()]}
        type="defaultSemiBold"
      >
        {props.ingredientName}
      </ThemedText>
    </TouchableOpacity>
  )
}

const styles = StyleSheet.create({
  buttonStyle: {
    paddingLeft: 15,
    width: "100%",
    height: 60,
    borderTopWidth: 1,
    borderBlockColor: "gray",
    display: "flex",
    flexDirection: "row",
    alignItems: "center",
  },
  completedBackground: {
    backgroundColor: "#121212",
  },
  uncompletedBackground: {
    backgroundColor: "#2e2e2e",
  },
  baseText: {
    left: 20,
    flex: 1,
  },
  completedText: {
    color: "#00400b",
    textDecorationLine: "line-through",
  },
  defaultText: {
    color: "white",
    textDecorationLine: "none",
  },
})
