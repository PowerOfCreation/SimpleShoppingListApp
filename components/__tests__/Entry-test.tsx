import * as React from "react"
import { render } from "@testing-library/react-native"
import { Entry, EntryProps } from "../Entry"

// Define default props
const defaultProps: EntryProps = {
  id: "1",
  ingredientName: "Default Ingredient",
  isCompleted: false,
  onToggleComplete: jest.fn(),
  onLongPress: jest.fn(),
  isEdited: false,
  onCancelEditing: jest.fn(),
  onSaveEditing: jest.fn(),
}

it(`renders correctly`, () => {
  const { toJSON } = render(<Entry {...defaultProps} ingredientName={""} />)

  expect(toJSON()).toMatchSnapshot()
})

it("applies correct style when isCompleted is true", () => {
  const { getByText } = render(
    <Entry {...defaultProps} ingredientName={"Tomato"} isCompleted={true} />
  )

  const entryText = getByText("Tomato")

  expect(entryText).toHaveStyle({
    textDecorationLine: "line-through",
  })
})

it("applies correct style when isCompleted is false", () => {
  const { getByText } = render(
    <Entry {...defaultProps} ingredientName={"Tomato"} isCompleted={false} />
  )

  const entryText = getByText("Tomato")

  expect(entryText).toHaveStyle({
    textDecorationLine: "none",
  })
})
