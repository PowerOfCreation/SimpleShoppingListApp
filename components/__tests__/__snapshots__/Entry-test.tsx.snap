// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#2e2e2e",
      "borderBlockColor": "gray",
      "borderTopWidth": 1,
      "display": "flex",
      "flexDirection": "row",
      "height": 60,
      "opacity": 1,
      "paddingLeft": 15,
      "width": "100%",
    }
  }
>
  <Text />
  <Text
    style={
      [
        {
          "color": "#11181C",
        },
        {
          "fontSize": 16,
          "fontWeight": "600",
          "lineHeight": 24,
        },
        [
          {
            "flex": 1,
            "left": 20,
          },
          [
            {
              "color": "white",
              "textDecorationLine": "none",
            },
          ],
        ],
      ]
    }
  />
</View>
`;
